internal/backend/local/test.go

```diff
@@ -1023,26 +1030,54 @@ func (runner *TestFileRunner) GetVariables(config *configs.Config, run *modulete
 		diags = diags.Append(valueDiags)
 	}
 
-	// Second, we'll check the run level variables.
+	// We need to also process all global variables for tests.
+	// If we run "terrafrom test" from the testing directory itself,
+	// the filepath.Dir(file.Name) is equal to "." so we need to extend the logic
+	if filepath.Dir(file.Name) == runner.Suite.TestingDirectory {
+		for name, value := range runner.Suite.GlobalTestVariables {
+			if !relevantVariables[name] {
+				// Then this run block doesn't need this value.
+				continue
+			}
 
-	// This is a bit more complicated, as the run level variables can reference
-	// previously defined variables.
+			// By default, we parse global variables as HCL inputs.
+			parsingMode := configs.VariableParseHCL
 
-	// Preload the available expressions, we're going to validate them when we
-	// build the context.
-	var exprs []hcl.Expression
-	for _, expr := range run.Config.Variables {
-		exprs = append(exprs, expr)
+			cfg, exists := config.Module.Variables[name]
```

Comment:
This is probably the result of cherry-picking from my v1.6 branch. Would be nice to review it before, as it has nothing with my changes